src/containers/App.js

```diff
@@ -0,0 +1,98 @@
+// src/containers/App.js
+
+import React, { Component } from 'react'
+
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS } from '../constants';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+import Todos from '../components/Todos';
+import Todo from '../components/Todo';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+
+		// JSX supports passing in element arrays
+		/*
+		const array = [
+			<Todo label="First todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Second todo..." deleteHandler={this.handleTodoDelete} />,
+			<Todo label="Third todo..." deleteHandler={this.handleTodoDelete} />
+		];
+		*/
+		this.filter = this.filter.bind(this);
+		this.handleTodoAdd = this.handleTodoAdd.bind(this);
+		this.handleTodoClear = this.handleTodoClear.bind(this);
+		this.handleTodoDelete = this.handleTodoDelete.bind(this);
+		this.handleTodoComplete = this.handleTodoComplete.bind(this);
+
+		this.state = {
+			lastIndex: 1,
+			show: ALL_TODOS,
+			items: []
+		};		
+	}
+
+	handleTodoAdd(e) {
+		console.log('add item');
+
+		if (e.target.value == '') return 0;
+
+    		if (e.key === 'Enter') {
+			this.setState({ lastIndex : this.state.lastIndex + 1 });
+      			this.setState({ items : this.state.items.concat(<Todo key={this.state.lastIndex} completed={false} label={e.target.value} deleteHandler={this.handleTodoDelete} completeHandler={this.handleTodoComplete}/>)});
```

Comment:
I would avoid storing React elements inside of a components state.  React components should live inside of the render function or within a function that is called by the render function.  

In this case you are storing Todo components within the App state and passing it to the Todos component via props.  I would prefer to represent the items in their simplest form (array of objects) and pass that array through to the Todos component.